src/server.ts

```diff
@@ -1,47 +1,103 @@
 import * as fsPath from 'path';
+import * as http from 'http';
 import * as nunjucks from 'nunjucks';
 
 import {Pod} from './pod';
 import {StaticRoute} from './router';
+import {Watcher} from './watcher';
 
 import express = require('express');
 
-export class Server {
-  constructor() {}
+interface ServerStartOptions {
+  log: boolean;
+  watch: boolean;
 }
 
-export function createApp(pod: Pod) {
-  const app = express();
-  app.disable('x-powered-by');
-  pod.plugins.trigger('createServer', app);
-  app.all('/*', async (req: express.Request, res: express.Response) => {
-    try {
-      const route = pod.router.resolve(req.path);
-      if (!route) {
-        res
-          .status(404)
-          .sendFile(fsPath.join(__dirname, './static/', 'error-no-route.html'));
-        return;
+export class Server {
```

Comment:
so i'm thinking about this api and it doesn't quite feel natural. e.g. it's weird that you pass in a port to the constructor instead of the `start()` method for example. the "watch" and "log" options also feel like internal options that an external user shouldn't care about. i think maybe we should take a step back and figure out how we want callers to be calling this class, e.g.:

```
const server = new Server(pod, /* do we want all the options here? */ {port: 8080, autoreload: true});
server.start();  // is this async?
// maybe the caller should be responsible for logging out the info
server.start(() => {
  console.log(`started on port xyz');
});
// and maybe the caller should be repsonsible for logging reloads also
server.on('reload', () => {
  console.log('server reloaded');
});
```

Reply:
Thanks for feedback, agreed it felt a little kludgy. I fixed it up and add a listener to output the initial console message on server start. The reload message is logged from within the watcher now.